core/jvm/src/main/scala/zio/interop/javaz.scala

```diff
@@ -46,9 +46,9 @@ private[zio] object javaz {
     }
 
   private def catchFromGet(isFatal: Throwable => Boolean): PartialFunction[Throwable, Task[Nothing]] = {
-    case e: CompletionException =>
+    case e: CompletionException if e.getCause.ne(null) =>
```

Comment:
Not sure it makes sense for us to handle this here. This introduces the possibility that the `Task` will fail with a `CompletionException` or an `ExecutionException` which weren't possibilities before and the caller needs to handle. The `null` value of the underlying cause is an issue that was created by whoever created the `Future` and potentially needs to be handled by the caller.

Reply:
I too am torn on this. The reason why I went ahead and implemented this is because I was worried that returning `Task.fail(null)` is even worse on our users than the current state.

But if you'd prefer to rather keep it like this, no worries, we can just close the PR :wink:

Reply:
I think it is probably better to leave it as is since the `null` value is a pre-existing issue and handling it creates new different and potentially unexpected states but it is a good thought.